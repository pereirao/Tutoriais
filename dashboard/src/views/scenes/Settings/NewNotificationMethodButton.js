// @flow

import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import I18n from '../../../localization';
import NewNotificationMethodModal from './NewNotificationMethodModal';
import MBButton from '../../elements/MBButton';
import MBIcon from '../../elements/MBIcon';
import { notification_method_actions } from '../../../business/notification_method';

type NewNotificationMethodButtonState = {is_modal_open: boolean};
export class NewNotificationMethodButton extends PureComponent {
  state: NewNotificationMethodButtonState

  constructor(props){
    super(props);
    this.state = {is_modal_open: false};
  }

  toggleModalVisibility = () => {
    this.props.resetForm();
    this.setState({ is_modal_open: !this.state.is_modal_open });
  }

  render(){
    return (
      <div>
        <MBButton
          title={I18n.t('accessibility.title.add_notification_method')}
          button_type="link"
          size="large"
          onClick={this.toggleModalVisibility}>
          <MBIcon icon="add" size="small" color="mb_red" />{I18n.t('ui.button.new_notification_method')}
        </MBButton>
        <NewNotificationMethodModal hidden={!this.state.is_modal_open} close={this.toggleModalVisibility} />
      </div>
    );
  }
}


const NewNotificationMethodFormDTP = {
  resetForm: notification_method_actions.resetNewNotificationMethodForm
};
const NewNotificationMethodButtonContainer = connect(null, NewNotificationMethodFormDTP)(NewNotificationMethodButton);

export default NewNotificationMethodButtonContainer;
