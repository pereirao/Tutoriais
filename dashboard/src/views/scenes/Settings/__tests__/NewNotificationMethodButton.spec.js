import React from 'react';

// NOTE: the named export isn't a connected component
import { NewNotificationMethodButton } from '../NewNotificationMethodButton';

it('renders correctly', () => {
  expect(shallow(<NewNotificationMethodButton />)).toMatchSnapshot();
});
