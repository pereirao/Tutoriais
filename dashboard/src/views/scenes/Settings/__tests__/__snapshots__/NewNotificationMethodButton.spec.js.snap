// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <MBButton
    button_type="link"
    onClick={[Function]}
    size="large"
    title="Add a New Notification Method"
  >
    <MBIcon
      color="mb_red"
      icon="add"
      size="small"
    />
    New Notification Method
  </MBButton>
  <NewNotificationMethodModal
    close={[Function]}
    hidden={true}
  />
</div>
`;
